packages/tupaia-web/src/features/Map/MapOverlaySelector/MapOverlayList.tsx

```diff
@@ -86,27 +88,45 @@ const MapOverlayAccordion = ({ mapOverlayGroup }: { mapOverlayGroup: MapOverlayG
           mapOverlay.children ? (
             <MapOverlayAccordion mapOverlayGroup={mapOverlay} key={mapOverlay.name} />
           ) : (
-            <FormControlLabel value={mapOverlay.code} control={<Radio />} label={mapOverlay.name} />
+            <FormControlLabel
+              value={mapOverlay.code}
+              control={<Radio />}
+              label={mapOverlay.name}
+              key={mapOverlay.code}
+            />
           ),
         )}
       </AccordionContent>
     </AccordionWrapper>
   );
 };
 
+const useOnChangeMapOverlay = () => {
+  const [urlSearchParams, setUrlParams] = useSearchParams();
+  const onChangeMapOverlay = (e: ChangeEvent<HTMLInputElement>) => {
+    urlSearchParams.set(URL_SEARCH_PARAMS.MAP_OVERLAY, e.target.value);
+    // when overlay changes, reset period to default
+    urlSearchParams.set(URL_SEARCH_PARAMS.MAP_OVERLAY_PERIOD, DEFAULT_PERIOD_PARAM_STRING);
+    setUrlParams(urlSearchParams);
+  };
+
+  return { onChangeMapOverlay };
+};
```

Comment:
I'm guessing this was moved to this file to be more contained in where it's used? In that case, I'm on the fence about whether it should be a hook or just a function inside the component

Reply:
yes it was because I think it will only be used in the one place and yes I think that you are right to move it in to the component.